HOW/tech/applications/cell-joining-code/cell-joining-code-v1.md

```diff
@@ -0,0 +1,110 @@
+---
+HOW: /tech/applications/cell-joining-code
+Source: /social/process/how/how-v1
+Status: Draft
+Created: 2024-11-23
+Version: idx/1
+---
+
+## Title
+
+Cell Joining Code
+
+## Authors
+
+@mattyg <matt@buildyourweb.app>, @zippy <eric@holochain.org>, Claude AI 3.5 Sonnet
+
+## Abstract
+
+This HOW proposes a standardized URI format for cell joining codes that enables conductors to join specific networks. The format encapsulates all necessary information for network joining in a single, shareable URI string, excluding the actual DNA.  The format also allows app-specific data to be carried so that UIs can display arbitrary data as needed for good 
+
+## Motivation
+
+Joining a Cell network requires multiple pieces of information to be shared: DNA hash, network seed, membrane proof, and app-specific metadata. Several apps have run into the need to share this information as a single string and implemented their own joining code format.
+
+A standardized joining code format would:
+- Simplify the network joining process
+- Simplify use of network joining codes in apps
+- Reduce user error during configuration
+- Enable one-click joining capabilities
+- Facilitate easier sharing of network access
+- Facilitate services providing "always-on" Cells: simply provide the DNA and joining code
```

Comment:
```suggestion
- Facilitate services providing "always-on" Cells: simply provide the .dna file and joining code (this does *not* include always-on nodes for happs).
```

The only use case I'm aware of for this is holo's always online node service. I don't know if they would be supported by this proposal since they may host the entire happ, not just a single cell.

Although maybe they do want to support hosting a single cell. If so, then I believe providing the `.dna` file along with the joining code would be sufficient.